src/WebApi/Application/UseCases/QuestionList/GetQuestionList/GetQuestionListUseCase.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Application.Repositories;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Application.UseCases.QuestionList.GetQuestionList;
+
+public class GetQuestionListUseCase : IGetQuestionListUseCase
+{
+    private IOutputPort _outputPort;
+
+    private readonly IQuestionListRepository _questionListRepository;
+
+    public GetQuestionListUseCase(IQuestionListRepository questionListRepository)
+    {
+        _outputPort = new GetQuestionListPresenter();
+        _questionListRepository = questionListRepository;
+    }
+
+    public async Task Execute(GetQuestionListInput input)
+    {
+        if (input == null)
+        {
+            throw new ArgumentNullException(nameof(input));
+        }
+
+        await GetQuestionListInternal(input);
+    }
+
+    public void SetOutputPort(IOutputPort outputPort) => _outputPort = outputPort;
+
+    private async Task GetQuestionListInternal(GetQuestionListInput input)
+    {
+        var questionLists = await _questionListRepository.Get(input);
+
+        if (questionLists != null && questionLists.Any())
+        {
+            _outputPort.Ok(questionLists);
+        }
+        else
+        {
+            _outputPort.NotFound();
+        }
```

Comment:
(just a suggestion) Another point where we technically could communicate irregular behavior first and leave the OK response till the end.

(just a draft, unverified code follows)
```cs
if (questionLists is null || !questionLists.Any())
{
    _outputPort.NotFound();
    return;
}

_outputPort.Ok(questionLists);
```

It's the small things that communicate intent a tiny bit better that makes the overall codebase a joy to work with.
Again, just a suggestion, small methods keep this from being a problem. 😄